client/coral-embed-stream/src/containers/Stream.js

```diff
@@ -112,7 +112,8 @@ class StreamContainer extends React.Component {
 
   componentDidMount() {
     if (this.props.previousTab) {
-      this.props.data.refetch();
+      this.props.data.refetch()
+        .then(() => this.props.setCommentCountCache(0));
```

Comment:
@cvle wrt setting to 0... that was silly. Yes, it did 'work' insofar as it hid the NewCount, but probably would have caused other badness. I think I read too fast and thought the commentCountCache reflected the gap between what the client knows about and the server has. Silly me, and good catch. I've updated to set it to the new asset.commentCount from the refetch.

> The commentCountCache basically should cache asset.commentCount from the last full query,

This is the type of invariant I was looking for. If this is the case, that we always want to reset commentCountCache to asset.comments whenever EmbedQuery is successful, then what do you think about handling this in the reducer that manages commentCountCache?

Could do so by adding the following to the [stream reducer](https://github.com/coralproject/talk/blob/master/client/coral-embed-stream/src/reducers/stream.js#L42)

```javascript
case 'APOLLO_QUERY_RESULT':
  switch (action.operationName) {
    case 'EmbedQuery':
      const commentCount = action.result && action.result.data && action.result.data.asset && action.result.data.asset.commentCount
      if (typeof commentCount === 'number') {
        return {
          ...state,
          commentCountCache: commentCount,
        }
      }
      break;
    default:
      break;
  }
```